collect_app/src/main/java/org/odk/collect/android/logic/PropertyManager.java

```diff
@@ -109,7 +108,7 @@ public PropertyManager(Context context) {
         initUserDefined(prefs, KEY_METADATA_EMAIL,       PROPMGR_EMAIL,         SCHEME_MAILTO);
     }
 
-    private IdAndPrefix findDeviceId(Context context, TelephonyManager telephonyManager) {
+    private IdAndPrefix findDeviceId(Context context, TelephonyManager telephonyManager) throws SecurityException {
```

Comment:
How about adding a comment describing that the security exception could be caused by attempting to get the device id without permissions but that the permission check is handled by the clients of this class? It might be helpful to refer to specifically link to https://github.com/opendatakit/collect/pull/2506 to provide context because the approach is certainly surprising (and could likely eventually be simplified).

Reply:
Thanks for the feedback! Yes, the way I have done it really hides the problem, doesn't it? Let me update. Thanks for the reference to #2506, that's very helpful.